Some documentation fixes (#1153)
ginkgo-project/ginkgo: INSTALL.md

+ GINKGO_BUILD_HWLOC=ON:
  [hwloc](https://www.open-mpi.org/projects/hwloc) to detect and control cores
  and devices.
+ GINKGO_BUILD_HWLOC=ON and GINKGO_BUILD_TESTS=ON:
  [libnuma](https://www.man7.org/linux/man-pages/man3/numa.3.html) is required
  when testing the functions provided through MachineTopology.
+ GINKGO_BUILD_EXAMPLES=ON:

>   [OpenCV](https://opencv.org/) is required for some examples.

Reviewer: ```suggestion
  [OpenCV](https://opencv.org/) is required for some examples, they are disabled when OpenCV is not available.
```

+ GINKGO_BUILD_DOC=ON:
  [doxygen](https://www.doxygen.nl/) is required to build the documentation and

>   additionally graphviz and dot are required to build the class hierarchy graphs.

Reviewer: dot is part of graphviz ;)
```suggestion
  additionally [graphviz](https://graphviz.org/) are required to build the class hierarchy graphs.
```

>   additionally [graphviz](https://graphviz.org/) are required to build the class hierarchy graphs.

Reviewer 2: ```suggestion
  additionally [graphviz](https://graphviz.org/) is required to build the class hierarchy graphs.
```
